### PR TITLE
feat: Make Stacked notifications feature public

### PR DESCRIPTION
**Pull request by jperals**
### Description

This change does the following:
- Merge the private API props for the Stacked notifications feature with the public Flashbar API
- Make changes to the new API prop docs according to content review
- Make changes to test utils docs, also content reviewed

### How has this been tested?

Documenter output was fed into a website build, albeit in a semi-manual process.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/flashbar/index.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function Flashbar(props: FlashbarProps | CollapsibleFlashbarProps
   }
 }
 
-function isStackedFlashbar(props: any): props is CollapsibleFlashbarProps {
+function isStackedFlashbar(props: FlashbarProps) {
```
Comment (gethinwebster) [inline src/flashbar/index.tsx:27]: is this needed anymore now that `stackItems` exists on the main interface?

**Review by gethinwebster — COMMENTED** [review]
how exactly did people use the 'beta' version of this? was it fully available in the source code & typings, but just hidden from the website API tab?

**Comment by jperals** [comment]
@gethinwebster Yes, quite like that. There were two interfaces, `FlashbarProps` and `CollapsibleFlashbarProps`. The Flashbar used `FlashbarProps | CollapsibleFlashbarProps` and only `FlashbarProps` is exported by our tooling in a way that it becomes _discoverable_.

File: src/flashbar/index.tsx (hunk 1 above)
Comment (jperals) [reply src/flashbar/index.tsx]: Just simplified this

**Review by gethinwebster — APPROVED** [review]

**Review by abdhalees — APPROVED** [review]

**Pull request merged**